osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/pushsetup/SetPushSetupAlarmCommandExecutor.java

```diff
@@ -65,45 +72,136 @@ public AccessResultCode execute(
       final MessageMetadata messageMetadata)
       throws ProtocolAdapterException {
 
-    final SetParameter setParameterSendDestinationAndMethod = this.getSetParameter(pushSetupAlarm);
+    this.checkPushSetupAlarm(pushSetupAlarm);
 
-    final AccessResultCode resultCode =
-        this.getAccessResultSetSendDestinationAndMethod(
-            "PushSetupAlarm", conn, OBIS_CODE, setParameterSendDestinationAndMethod);
+    AccessResultCode resultCode = null;
+
+    if (pushSetupAlarm.hasSendDestinationAndMethod()) {
+      resultCode = this.setSendDestinationAndMethod(conn, pushSetupAlarm);
+
+      if (resultCode != AccessResultCode.SUCCESS) {
+        return resultCode;
+      }
+    }
+
+    if (pushSetupAlarm.hasPushObjectList()) {
+      resultCode = this.setPushObjectList(conn, pushSetupAlarm);
 
-    if (resultCode != null) {
       return resultCode;
+    }
+
+    if (resultCode == null) {
```

Comment:
I think you should do this check before the first return without conditions on line 90 to prevent a null return.
Or line 90 should have the same conditions as line 82.

Reply:
This code is more complex then needed. At line 93 resultCode will never be null, because both `setSendDestinationAndMethod` and `setPushObjectList` will throw an exception if the result is null. I refactored the code to make it more simple.